presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -632,4 +644,57 @@ public Page filterPageToEligibleRowsOrDiscard(Page page)
             return page.getPositions(ids.elements(), 0, retainedRowCount);
         }
     }
+
+    public static class BucketValidator
+    {
+        private final Path path;
+        private final int[] bucketColumnIndices;
+        private final List<TypeInfo> bucketColumnTypes;
+        private final BucketingVersion bucketingVersion;
+        private final int bucketCount;
+        private final int expectedBucket;
+
+        public BucketValidator(
+                Path path,
+                int[] bucketColumnIndices,
+                List<TypeInfo> bucketColumnTypes,
+                BucketingVersion bucketingVersion,
+                int bucketCount,
+                int expectedBucket)
+        {
+            this.path = requireNonNull(path, "path is null");
```

Comment:
check length of `bucketColumnIndices` and length of `bucketColumnTypes` are the same

Reply:
Done